lib/rdl/types/var.rb

```diff
@@ -121,22 +121,19 @@ def add_lbound(typ, ast, new_cons = {}, propagate: false)
     end
 
     def to_s # :nodoc:
-      if @to_infer
-        if @solution
-          return @solution.to_s
```

Comment:
@jeffrey-s-foster This was causing it to go into an infinite loop---the solutions seem to reference the methods they came from.

Reply:
@mckaz, can you look at this?

Reply:
Apologies, just got around to seeing this. The the `@solution` field was something I added a while back but then decided not to use, so it was always nil before. Now that it's being used again it started causing this error.

Geoff and I were just chatting, we decided to leave `@solution` out from the `to_s` method entirely.